README.md

```diff
@@ -60,14 +59,82 @@ python3 -m pytest --cov-report=term-missing --cov=um2nc
 > [!TIP]
 > To generate an HTML coverage report substitute `term-missing` with `html`.
 
-## User documentation
+## Usage instructions
 
-TODO: this needs to cover:
+`um2nc` utilities for converting Unified Model files to netCDF can be accessed through the command line or as a `Python3` API. This user documentation details the available command line utilities:
+* [`um2nc`](#um2nc)
+* [`esm1p5_convert_nc`](#esm1p5_convert_nc)
 
-1. Running `um2netcdf` standalone
-2. Using the workflow run script
-3. Using `um2netcdf` as an API
+### `um2nc`
+The `um2nc` command converts a single Unified Model data file `infile` to a netCDF file `outfile`, and is used as follows:
+```
+um2nc [options] infile outfile
+```
+The following options are available for configuring the conversion:
+
+**User information options**:
+* `-h, --help` Display a help message and exit.
+* `-v, --verbose`  Display verbose output (use `-vv` for the highest level of output).
+
+**Output file format options**:
+* `-k NC_KIND` netCDF output format. Options `1`: classic, `2`: 64-bit offset, `3`: netCDF-4, `4`: netCDF-4 classic model. Default `3`.
+* `-c COMPRESSION` netCDF compression level (`0`=none, `9`=max). Default `4`.
+* `--64` Write 64 bit output when input is 64 bit. When absent, output will be 32 bit.
+
+**Variable selection options**:
+
+* `--include ITEM_CODE_1 [ITEM_CODE_2 ...]` List of variables to include in the output, specified by their item codes. Item codes are given in the form `1000 * section number + item number`. Any other variables present in `infile` will not be written to `outfile`.
+* `--exclude ITEM_CODE_1 [ITEM_CODE_2 ...]` List of variables to exclude from the output, specified by their item codes. Item codes are given in the form `1000 * section number + item number`. All other variables present in `infile` will be written to `outfile`.
+
+The options `--include` and `--exclude` cannot be used simultaneously. When neither are present, all variables in `infile` will be written to `outfile`.
+
+**Masking options**:
+
+Points on a pressure level grid may fall below ground-level for some of the duration represented by a data point in the input `infile`. When heaviside masking is enabled, pressure level data at a given location will be masked if it fell below ground-level for longer than a critical fraction (`HCRIT`) of the time.
+
+By default, variables on pressure level grids will be masked by the appropriate heaviside variable found in `infile`. If the heaviside variable cannot be found, these variables will be omitted from the output `outfile`. This behaviour is modified by the following options:
+
+* `--hcrit HCRIT` Critical value of heaviside variable for pressure level masking. Has no effect if the required heaviside variable is missing, or if `--nomask` is selected. Default `0.5`.
```

Comment:
Good question – I just double checked to make sure what I'd written was correct, fortunately it is!
https://github.com/ACCESS-NRI/UM7/blob/457d1da26332a78a2418c888f7f74d71c81f69bd/umbase_hg3/src/atmosphere/climate_diagnostics/eot_diag.F90#L2333-L2339

Reply:
Good to know! I had no idea of what exactly that would mean :D 
Then I think my suggestion stays.

Reply:
I just double checked how the masking is applied, and realised my interpretation of `HCRIT` was actually incorrect. 

`HCRIT` is the fraction of time that a point must be _above_ ground for its value to be kept, and it will be masked otherwise. I will update the paragraph below `_Masking options:_` to clarify this.

I think it would be good to make it clear in the usage text that the "time extent" refers to individual data points, rather than whole input variables, i.e. the masking is applied on a point by point basis, dependent on the amount of time that each point spent above ground-level.

What would you think of the following:

```
--hcrit HCRIT` Minimum fraction of the time spent above ground-level for a pressure
grid data point to be considered valid.  Data points in pressure grid variables
will be masked if they fell above ground-level for less than
the critical fraction `HCRIT` of the time. This option has no effect
when used together with the `--nomask` option. Default `0.5`.
```

Reply:
Ah, ok it makes sense. 
So from what I can understand the `heaviside` variable has values from 0 to 1 corresponding to the fraction of the time the pressure level fields were above the surface. 
So, when setting `HCRIT`, points are masked when `heaviside < HCRIT`. 
Correct? 

In that case a minor modification to your updated text:
```diff
- `--hcrit HCRIT` Minimum fraction of the time spent above ground-level for a pressure grid data point to be considered valid.  Data points in pressure grid variables will be masked if they fell above ground-level for less than the critical fraction `HCRIT` of the time. This option has no effect when used together with the `--nomask` option. Default `0.5`.
+ `--hcrit HCRIT` Minimum fraction of the time spent above ground-level for a pressure grid data point to be considered valid.  Data points in pressure grid variables will be masked if they were above ground-level for less than the critical fraction `HCRIT` of the time. This option has no effect when used together with the `--nomask` option. Default `0.5`.
```

Reply:
>So, when setting HCRIT, points are masked when heaviside < HCRIT.
Correct?

Yeah exactly! This masking is done here in `um2nc`:

https://github.com/ACCESS-NRI/um2nc-standalone/blob/cc0f2ba64aa5c6890b263da4f024c187632eb63d/src/um2nc/um2netcdf.py#L501

Good suggestion, I've made that change and fixed the preceding paragraph in `bb6ceda`.

Reply:
Yes, I saw that bit to understand better how the masking was performed :D 

Perfect! Thank you!